packages/svelte/src/compiler/phases/2-analyze/visitors/RenderTag.js

```diff
@@ -49,8 +53,31 @@ export function RenderTag(node, context) {
 	) {
 		e.render_tag_invalid_call_expression(node);
 	}
+	
+	const parent = context.path.at(-2);
+	const is_animated = snippet?.body.nodes.some(is_animate_directive);
+
+	if(is_animated) {
+		if (parent?.type !== 'EachBlock') {
+			e.animation_invalid_placement(node);
+		}
+		else if (!parent.key) {
+			e.animation_missing_key(parent);
+		}
+	}
 
 	mark_subtree_dynamic(context.path);
 
 	context.next({ ...context.state, render_tag: node });
 }
+
+/** @param {AST.Text | AST.Tag | AST.ElementLike | AST.Comment | AST.Block} child */
+function is_animate_directive(child) {
```

Comment:
This function is used twice as it is also in [3-transform/client/visitors/EachBlock.js line 347](https://github.com/sveltejs/svelte/pull/14796/files#diff-9babb73be31dc801fa873f90634a31ed122d7393f11a123825652ce6900d5a82R347). Do I put it anywhere to re-use it across the two phases?